libyul/backends/evm/SSACFGLiveness.h

```diff
@@ -35,19 +33,88 @@ namespace solidity::yul
 class SSACFGLiveness
 {
 public:
-	using LivenessData = std::set<SSACFG::ValueId>;
+	class LivenessData
+	{
+	public:
+		using Count = std::uint32_t;
+		using Value = SSACFG::ValueId;
+		using LiveCounts = std::vector<std::pair<Value, Count>>;
+
+		LivenessData() = default;
+		template<std::input_iterator Iter, std::sentinel_for<Iter> Sentinel>
+		LivenessData(Iter begin, Sentinel end): m_liveCounts(begin, end) {
+		}
+		explicit LivenessData(LiveCounts&& _liveCounts): m_liveCounts(std::move(_liveCounts)) {}
+
+		bool contains(SSACFG::ValueId const& _valueId) const;
+		Count count(SSACFG::ValueId const& _valueId) const;
```

Comment:
```suggestion
		bool contains(Value const& _valueId) const;
		Count count(Value const& _valueId) const;
```

Since you have the `using`, it would be nice to consistently use it.